runpod/serverless/modules/worker_state.py

```diff
@@ -40,30 +22,36 @@ def get_auth_header():
     return {"Authorization": f"{os.environ.get('RUNPOD_AI_API_KEY')}"}
 
 
-def get_current_job_id():
-    '''
-    Returns the current job id.
-    '''
-    return CURRENT_JOB_ID
 
+class Jobs:
+    ''' Track the state of current jobs.'''
 
-def get_done_url():
-    '''
-    Constructs the done URL using the current job id.
-    '''
-    return JOB_DONE_URL_TEMPLATE.replace('$ID', CURRENT_JOB_ID)
+    _instance = None
+    jobs = []
```

Comment:
Looks good! One precondition for proper execution of this code region is the assumption that JobTake will always return unique job_id values (e.g. it'll never return the same job_id more than once). Otherwise, it can lead to problems with add_job, remove_job, and the information sent via our heartbeat.

Violation of the pre-condition could result in adding the same job_id twice within a window. Calling remove_job once would delete both job_ids, and calling it the second time would cause a ValueError exception. Additionally, a race condition would be possible, where the list of jobs sent via ping can have the same job_id repeated twice.

If JobTake is guaranteed to always returns a job_id at most once to a single worker, then we can safely ignore above!